conn_pool.go

```diff
@@ -105,7 +112,24 @@ func (p *ConnPool) Acquire() (c *Conn, err error) {
 		if p.logLevel >= LogLevelWarn {
 			p.logger.Warn("All connections in pool are busy - waiting...")
 		}
+
+		deadlineReached := false
+
+		if p.connAllocTimeout > 0 {
+			timer := time.NewTimer(p.connAllocTimeout)
+			defer timer.Stop()
+
+			select {
+			case <-timer.C:
+				deadlineReached = true
+				p.cond.Signal()
+			}
```

Comment:
won't this simply block until `timer` times out? does it need another `case` to exit early if a connection can be acquired?
maybe it's supposed to run on a separate goroutine?